pkg/adaptor/cloud/vsphere/provider.go

```diff
@@ -49,6 +54,27 @@ func NewProvider(config *Config) (cloud.Provider, error) {
 	return provider, nil
 }
 
+func checkConfig(config *Config) error {
+
+	// Do some initial checks of the optional input values
+
+	if config.DRS == "true" {
+		if config.Cluster == "" {
+			return fmt.Errorf("Error: A cluster name is required with DRS")
+		}
+		return nil
+	}
+
+	// Hosts and Datastores are required when not using DRS
+	// We cannot check for a cluster as hosts do not have to be part of a cluster.
+
+	if config.Host == "" || config.Datastore == "" {
+		return fmt.Errorf("Error: A host and datastore name are required")
```

Comment:
Are both host and datastore needed or either of them?

Reply:
Both are required so if either of them are empty it is an error.